doc/devdocs/common/FilePreviewCommon.md

```diff
@@ -0,0 +1,70 @@
+
+This previewer is used for the File Explorer Dev File Previewer, as well as PowerToys Peek.
+
+### Update Monaco Editor
+
+1. Download Monaco editor with [npm](https://www.npmjs.com/): Run `npm i monaco-editor` in the command prompt.
+2. Delete everything except the `min` folder (the minimised code) from the downloaded files.
+3. Copy the `min` folder into the `src/common/FilePreviewCommon/monacoSRC` folder of the PowerToys project.
+4. Generate the JSON file as described in the generate [monaco_languages.json file](#monaco_languagesjson) section.
+
+### Add a new language definition
+
+As an example on how to add a new language definition you can look at the one for [registry files](/src/common/FilePreviewCommon/Assets/Monaco/customLanguages/reg.js).
+
+1. Add the language definition (written with [Monarch](https://microsoft.github.io/monaco-editor/monarch.html)) to the [folder containing Monaco custom languages](/src/common/FilePreviewCommon/Assets/Monaco/customLanguages/). The file should be formatted like in the example below. (Please change `idDefinition` to the name of your language.)
+
+```javascript
+export function idDefinition() {
+    return {
+        ...
+    }
+}
+```
+
+2. Add the following line to the [`monacoSpecialLanguages.js`](/src/common/FilePreviewCommon/Assets/Monaco/monacoSpecialLanguages.js) file:
+
+```javascript
+import { idDefinition } from './customLanguages/file.js';
+```
+
+3. In the [`monacoSpecialLanguages.js`](/src/common/FilePreviewCommon/Assets/Monaco/monacoSpecialLanguages.js) file add the following line to the `registerAdditionalLanguages` function:
+
+```javascript
+registerAdditionalNewLanguage("id", [".fileExtension"], idDefinition(), monaco)
+```
+
+  * The id can be anything. Recommended is one of the file extensions. For example "php" or "reg".
+4. Copy the existing language definition into the `languageDefinitions` function in the same file. You can find the existing definitions in the following folder: [`/src/common/FilePreviewCommon//Assets/Monaco/monacoSRC/min/vs/basic-languages/`](/src/common/FilePreviewCommon/Assets/Monaco/monacoSRC/min/vs/basic-languages/).
+
+5. Execute the steps described in the [monaco_languages.json](#monaco_languagesjson) section.
+
+### Add a new file extension to an existing language
+
+1. In the [`monacoSpecialLanguages.js`](/src/common/FilePreviewCommon/Assets/Monaco/monacoSpecialLanguages.js) file add the following line to the `registerAdditionalLanguages` function. (`existingId` is the id of the language you want to add the extension to. You can find these id's in the [`monaco_languages.json`](/src/common/FilePreviewCommon/monaco_languages.json) file):
+
+```javascript
+registerAdditionalLanguage("id", [".fileExtension"], "existingId", monaco)
+```
+
+  * If for instance you want to add more extensions to the php language set the id to `phpExt` and the existingId to `php`.
```

Comment:
```suggestion
  * If for instance you want to add more extensions to the php language set the id to `newPhpExt` and the existingId to `php`.
```

Reply:
Why? just adding ext is what is done by other examples.

Reply:
My intention was to make clear that this is the new extension (that is linked to the old one).

Reply:
I think it is already pretty clear